js/ui/iconGrid.js

```diff
@@ -1185,8 +1187,16 @@ const PaginatedIconGrid = new Lang.Class({
     },
 
     adaptToSize: function(availWidth, availHeight) {
+        let size_changed = availWidth != this._adaptedWidth || availHeight != this._adaptedHeight;
+
         this.parent(availWidth, availHeight);
         this._computePages(availWidth, availHeight);
+
+        // If the adapted size changed, we must queue a relayout in the sizing
+        // machinery to make sure we always get the correct size. This cannot
+        // be done inside adaptToSize(), so we can't just call allocate() here.
+        if (size_changed)
+            this._grid.queue_relayout();
```

Comment:
IIRC, this was a speculative commit to try to fix the issues during startup, but my testing today found out that this is not really needed, please drop it.